src/tiled/objectselectionitem.cpp

```diff
@@ -199,16 +199,25 @@ void MapObjectLabel::syncWithMapObject(const MapRenderer &renderer)
     QPointF screenPos = renderer.pixelToScreenCoords(mObject->position());
     QRectF bounds = mObject->screenBounds(renderer);
 
-    // Adjust the bounding box for object rotation
-    bounds = rotateAt(screenPos, mObject->rotation()).mapRect(bounds);
+    if (mObject->shape() == MapObject::Point) {
+        // Use screenspace offset, since the Point annotation rescales with the zoom-level
+        mScreenspaceOffset = { 0, -bounds.height() };
 
-    // Center the object name on the object bounding box
-    QPointF pos((bounds.left() + bounds.right()) / 2, bounds.top());
+        setPos(screenPos);
+    } else {
+        mScreenspaceOffset = { 0, 0 };
 
-    if (auto mapScene = static_cast<MapScene*>(scene()))
-        pos += mapScene->absolutePositionForLayer(*mObject->objectGroup());
+        // Adjust the bounding box for object rotation
+        bounds = rotateAt(screenPos, mObject->rotation()).mapRect(bounds);
+
+        // Center the object name on the object bounding box
+        QPointF pos((bounds.left() + bounds.right()) / 2, bounds.top());
+
+        if (auto mapScene = static_cast<MapScene*>(scene()))
+            pos += mapScene->absolutePositionForLayer(*mObject->objectGroup());
```

Comment:
This call is here to take into account layer offsets as well as the parallax offset. It should be taken into account also for point objects.

Reply:
@bjorn ah okay, don't know how I missed this.  I tried it out with and without the `absolutePositionForLayer` added, it is working better now after your suggestion.  Thanks!